### ISSUE TITLE
SummaryJavadoc: false positive with @inheritDoc

### ISSUE THREAD

**Issue by dmitry-timofeev** [checkstyle/checkstyle#4766]
/var/tmp $ javac TestMap.java

/var/tmp $ cat TestMap.java
```
interface MyMap {
    /**
    * Puts a new key-value pair into the map. If this map already contains
    * a mapping for the specified key, overwrites the old value with the specified value.
    *
    * @param key a storage key
    * @param value a storage value to associate with the key
    * @throws IllegalArgumentException if some property of the key or the value prevents it
    *                                  from being stored in this map
    */
    void put(int key, int value);
}

public class TestMap implements MyMap {
    
    /**
    * {@inheritDoc}
    *
    * @param key a proof map key, must be non-negative
    * @param value a storage value to associate with the key, must be a power of two
    * @throws IllegalArgumentException if the size of the key is negative; or the value is not a power of two
    */
    public void put(int key, int value) {
        // no-op
    }
}
```
/var/tmp $ cat config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <property name="severity" value="error"/>

    <module name="TreeWalker">
        <module name="SummaryJavadoc">
            <property name="forbiddenSummaryFragments" value="^@return the *|^This method returns |^A [{]@code [a-zA-Z0-9]+[}]( is a )"/>
        </module>
    </module>
</module>
```

/var/tmp $ java -jar checkstyle-8.0-all.jar -c config.xml TestMap.java
```
Starting audit...
[ERROR] /home/dmitry/Documents/checkstyle-bug/TestMap.java:17: First sentence of Javadoc is incomplete (period is missing) or not present. [SummaryJavadoc]
Audit done.
Checkstyle ends with 1 errors.
```
---------------

Almost the same as #2805: I expect to see _no_ errors, since the inherited Javadoc has a correct first sentence.

--------------

**Comment by romani** [comment]
It is the same as recent #4744.
I am closing this as duplicate

**Comment by dmitry-timofeev** [comment]
@romani, I see, haven't searched for open issues properly, thanks.
